src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/MenuItemAutomationPeer.cs

```diff
@@ -104,7 +104,10 @@ override protected int GetSizeOfSetCore()
 
                 foreach (var item in parent.Items)
                 {
-                    if (item is Separator)
+                    // ItemAutomationPeer.GetSizeOfSetFromItemsControl excludes invisible UI elements
+                    // from its count, so we only exclude separators that are visible. Otherwise, we would
+                    // double count invisible separators, leading to a lower count than expected.
+                    if (item is Separator { Visibility: Visibility.Visible })
```

Comment:
[IsVisible](https://learn.microsoft.com/en-us/dotnet/api/system.windows.uielement.isvisible) sounds like better firt for this purpose

Reply:
(applies to all the visibiltiy tests)